### PR TITLE
Java latest version plugin app service updated

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/appservice/javaVersion.spec.js [hunk 1]
```diff
@@ -107,15 +119,10 @@ describe('javaVersion', function() {
         it('should give unknown result if app has no configs', function(done) {
             const cache = createErrorCache('configs');
             javaVersion.run(cache, {}, (err, results) => {
-                expect(results.length).to.equal(2);
+                expect(results.length).to.equal(1);
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.spec.js:122]:
```suggestion
                expect(results.length).to.equal(2);
```

File: plugins/azure/appservice/javaVersion.spec.js [hunk 2]
```diff
@@ -107,15 +119,10 @@ describe('javaVersion', function() {
...
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.spec.js:123]:
```suggestion
                expect(results.length).to.equal(1);
```

File: plugins/azure/appservice/javaVersion.spec.js [hunk 3]
```diff
@@ -108,14 +120,8 @@ describe('javaVersion', function() {
             const cache = createErrorCache('configs');
             javaVersion.run(cache, {}, (err, results) => {
                 expect(results.length).to.equal(2);
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.spec.js:122]:
```suggestion
                expect(results.length).to.equal(1);
```

File: plugins/azure/appservice/javaVersion.js [hunk 4]
```diff
@@ -49,29 +47,53 @@ module.exports = {
...
+                        let version = appConfig.linuxFxVersion;
+                        currentVersion = appConfig.linuxFxVersion.substring(version.indexOf('|')+1, version.lastIndexOf('-'));
+                        if (currentVersion && currentVersion != '' && currentVersion >= config.latestJavaVersion){
+                            versionAvailable = true;
+                        }
+                    }
+                }
+                if (!found){
+                    helpers.addResult(results, 0, 'No App Services with Java found', location);
+                    continue;
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:90]: use the found flag outside the for loop (as it is changed in php version)

Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:90]: use the found flag outside the for loop (as it is changed in php version)

File: config_example.js [hunk 5]
```diff
@@ -32,10 +32,6 @@ module.exports = {
             // OPTION 1: If using a credential JSON file, enter the path below
             // credential_file: '/path/to/file.json',
             // OPTION 2: If using hard-coded credentials, enter them below
```
Comment (alphadev4) [inline config_example.js:34]:
```suggestion
            // OPTION 2: If using hard-coded credentials, enter them below
            // application_id: process.env.AZURE_APPLICATION_ID || '',
            // key_value: process.env.AZURE_KEY_VALUE || '',
            // directory_id: process.env.AZURE_DIRECTORY_ID || '',
            // subscription_id: process.env.AZURE_SUBSCRIPTION_ID || ''
```

Comment (alphadev4) [inline config_example.js:34]:
```suggestion
            // OPTION 2: If using hard-coded credentials, enter them below
            // application_id: process.env.AZURE_APPLICATION_ID || '',
            // key_value: process.env.AZURE_KEY_VALUE || '',
            // directory_id: process.env.AZURE_DIRECTORY_ID || '',
            // subscription_id: process.env.AZURE_SUBSCRIPTION_ID || ''
```

File: plugins/azure/appservice/javaVersion.js [hunk 6]
```diff
@@ -48,29 +46,50 @@ module.exports = {
...
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                }       
+            }
 
-            if (!found) {
+            if (!found){
                 helpers.addResult(results, 0, 'No App Services with Java found', location);
             }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:94]:
```suggestion
```

File: plugins/azure/appservice/javaVersion.js [hunk 7]
```diff
@@ -48,29 +46,50 @@ module.exports = {
...
+                    }
                 }
-            });
+                if (found === true){
+                    if (versionAvailable && versionAvailable === true) {
+                        helpers.addResult(results, 0, `The Java version (${currentVersion}) is the latest version`, location, webApp.id);
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:89]:
```suggestion
                } else 
                helpers.addResult(results, 0, 'No App Services with Java found', location);
```

File: plugins/azure/appservice/javaVersion.js [hunk 8]
```diff
@@ -48,32 +46,52 @@ module.exports = {
...
+                if (found === true){
+                    if (versionAvailable && versionAvailable === true) {
+                        helpers.addResult(results, 0, `The Java version (${currentVersion}) is the latest version`, location, webApp.id);
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                } else {
+                      helpers.addResult(results, 0, 'No App Services with Java found', location);
+                }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:91]:
```suggestion
                }
```

File: plugins/azure/appservice/javaVersion.js [hunk 9]
```diff
@@ -48,32 +46,50 @@ module.exports = {
...
+                        }
+                    }
+                }
+                if (found === true){
+                    if (versionAvailable && versionAvailable === true) {
+                        helpers.addResult(results, 0, `The Java version (${currentVersion}) is the latest version`, location, webApp.id);
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:89]:
```suggestion
                }
                 if(!found) {
                      helpers.addResult(results, 0, 'No App Services with Java found', location);
                }
```

File: plugins/azure/appservice/javaVersion.js [hunk 10]
```diff
@@ -48,32 +46,53 @@ module.exports = {
...
+                 if(!found) {
+                      helpers.addResult(results, 0, 'No App Services with Java found', location);
+                }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:92]:
```suggestion
```

File: plugins/azure/appservice/javaVersion.js [hunk 11]
```diff
@@ -48,32 +46,53 @@ module.exports = {
...
             }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:94]:
```suggestion
        if(!found) {
                      helpers.addResult(results, 0, 'No App Services with Java found', location);
                }
```

File: plugins/azure/appservice/javaVersion.js [hunk 12]
```diff
@@ -48,31 +46,51 @@ module.exports = {
...
+                }
+                if (found === true){
+                    if (versionAvailable && versionAvailable === true) {
+                        helpers.addResult(results, 0, `The Java version (${currentVersion}) is the latest version`, location, webApp.id);
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                }
             }
+        if(!found) {
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:91]:
```suggestion
            if(!found) {
```

File: plugins/azure/appservice/javaVersion.js [hunk 13]
```diff
@@ -48,31 +46,51 @@ module.exports = {
...
+                      helpers.addResult(results, 0, 'No App Services with Java found', location);
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:92]:
```suggestion
                helpers.addResult(results, 0, 'No App Services with Java found', location);
```

File: plugins/azure/appservice/javaVersion.js [hunk 14]
```diff
@@ -48,31 +46,51 @@ module.exports = {
...
+                }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:93]:
```suggestion
          }
```

File: plugins/azure/appservice/javaVersion.js [hunk 15]
```diff
@@ -48,31 +46,51 @@ module.exports = {
...
+                        helpers.addResult(results, 0, `The Java version (${currentVersion}) is the latest version`, location, webApp.id);
+                    } else {
+                        helpers.addResult(results, 2, `The Java version (${currentVersion}) is not the latest version`, location, webApp.id);
+                    } 
+                }
+            }
             if (!found) {
                 helpers.addResult(results, 0, 'No App Services with Java found', location);
-            }
+          }
```
Comment (alphadev4) [inline plugins/azure/appservice/javaVersion.js:93]:
```suggestion
            }
```

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/azure/appservice/javaVersion.js [hunk 16]
```diff
@@ -48,28 +46,48 @@ module.exports = {
...
+                        found = true;
+                        let version = appConfig.linuxFxVersion;
+                        currentVersion = appConfig.linuxFxVersion.substring(version.indexOf('|')+1, version.lastIndexOf('-'));
+                        if (currentVersion && currentVersion != '' && parseFloat(currentVersion) >= parseFloat(config.latestJavaVersion)){
+                            versionAvailable = true;
+                        }
+                    }
+                }
+                if (found === true){
+                    if (versionAvailable && versionAvailable === true) {
```
Comment (AkhtarAmir) [inline plugins/azure/appservice/javaVersion.js:84]:
```suggestion
                if (found){
                    if (versionAvailable) {
```

**Review by AkhtarAmir — APPROVED** [review]

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**